File: encrypt.py
from Crypto.Cipher import AES
import base64


class Encrypt:
    def __init__(self, key: str) -> None:
        self.key = key.encode('utf-8')

    @staticmethod
    def pkcs7_padding(text: str) -> str:
        """明文使用 PKCS7 填充。"""
        bs = 16
        padding = bs - len(text.encode('utf-8')) % bs
        return text + chr(padding) * padding

    def aes_encrypt(self, content: str) -> str:
        """AES-ECB 加密，返回 base64 字符串。"""
        cipher = AES.new(self.key, AES.MODE_ECB)
        content_padding = self.pkcs7_padding(content)
        encrypt_bytes = cipher.encrypt(content_padding.encode('utf-8'))
        return str(base64.b64encode(encrypt_bytes), encoding='utf-8')


def aes_encrypt(text: str) -> str:
    """用内置密钥对明文做 AES 加密（登录密码用）。"""
    key = 'MWMqg2tPcDkxcm11'  # 密钥
    a = Encrypt(key=key)
    return a.aes_encrypt(text)

File: src/encrypt.rs
use aes::cipher::{generic_array::GenericArray, BlockEncrypt, KeyInit};
use aes::Aes128;
use base64::{engine::general_purpose::STANDARD, Engine};

/// 与 Python 版 encrypt.AES_encrypt 保持一致：
/// 密钥 MWMqg2tPcDkxcm11，AES-128-ECB，PKCS7 填充，base64 输出。
const KEY: &[u8; 16] = b"MWMqg2tPcDkxcm11";

pub fn aes_encrypt(plaintext: &str) -> String {
    let cipher = Aes128::new_from_slice(KEY).expect("AES key must be 16 bytes");

    let mut data = plaintext.as_bytes().to_vec();
    let padding = 16 - (data.len() % 16);
    data.extend(std::iter::repeat_n(padding as u8, padding));

    let mut encrypted = Vec::with_capacity(data.len());
    for chunk in data.chunks(16) {
        let mut block = GenericArray::clone_from_slice(chunk);
        cipher.encrypt_block(&mut block);
        encrypted.extend_from_slice(&block);
    }
    STANDARD.encode(encrypted)
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn known_vectors() {
        let cases = [
            ("zyx/020305", "5ZTBUxmD+OY7LL1nzUUz+g=="),
            ("123456", "OSfRhnd673K1Lp6cP4L6nA=="),
            ("", "mkpyTarWC0ro2N4QUBrjAQ=="),
            ("a", "hj0T3YA9PCHq7PAsPke1mQ=="),
            (
                "password12345678",
                "bLI9j6Y3UbBljQs6YgYfnJpKck2q1gtK6NjeEFAa4wE=",
            ),
            ("中文密码abc", "0sLEDhp4IRDqt9Y0z1Flvw=="),
        ];
        for (plain, expected) in cases {
            assert_eq!(aes_encrypt(plain), expected, "plaintext: {plain:?}");
        }
    }
}
